esx_service/vmci/vmci_client.c

```diff
@@ -360,7 +360,7 @@ Vmci_GetReply(int port, const char* json_request, const char* be_name,
       return CONN_FAILURE;
    }
 
-   req.mlen = strnlen(json_request, MAXBUF) + 1;
+   req.mlen = strnlen(json_request, MAXBUF-1) + 1;
```

Comment:
Great this fixes the issue but I am not sure if I understand the fix. Original code had "+1" to accomodate terminating null byte. Does it mean either c-go or vmci has issues when buffers crosses 1M boundry and another way to fix is make `MAXBUF = (1024 * 1024) - 1 `?